main.go

```diff
@@ -85,6 +86,11 @@ func isValidFunctionForDeploy(f Function) bool {
 		return false
 	}
 
+	if f.AllowUnauthenticated != "" && f.AllowUnauthenticated != "true" {
```

Comment:
This will prevent people from using `false`.

What about checking for ` != "" && != "true" && != "false"` and the and down in line 273 you check for ` == "true`?